app/components/Footer.jsx

```diff
@@ -35,10 +43,31 @@ export default class Footer extends Component {
     }
   }
 
+  toggleIsThemesBarOpen() {
+    if (this.state.isThemesBarOpen === true) {
+      this.setState({
+        isThemesBarOpen: false,
+      });
+    }
+    else {
+      this.setState({
+        isThemesBarOpen: true,
+      });
+    }
+  }
+
   render() {
     let path = '';
 
-    if (this.props.metadata) {
+    if (this.props.path) {
+      path = this.props.path;
+    }
+    else if (this.props.metadata) {
+      if(this.props.metadata.firstname){
+        path = buildPath(`${this.props.metadata.name} ${this.props.metadata.firstname}`);
+      }else {
+        path = buildPath(this.props.metadata.name);
+      }
```

Comment:
This logic is getting complex and I think it would be worth it to extract it into its own function that returns the path.